engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/windowconsumer/RingBufferWindowConsumer.java

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.updateby.rollingformulamulticolumn.windowconsumer;
+
+import io.deephaven.base.ringbuffer.*;
+import io.deephaven.chunk.Chunk;
+
+public abstract class RingBufferWindowConsumer {
+    public static RingBufferWindowConsumer create(final RingBuffer buffer) {
```

Comment:
You can add some javadoc for the class and methods since its not immediately obvious from the name, like in `RingeBufferVectorWrapper`.